include/flatbuffers/base.h

```diff
@@ -306,7 +306,7 @@ typedef uintmax_t largest_scalar_t;
   #pragma warning(disable: 4127) // C4127: conditional expression is constant
 #endif
 
-template<typename T> T EndianSwap(T t) {
+template<typename T> T EndianSwap(const T &t) {
```

Comment:
This change worries me.. this function is meant for scalars only, and if any compiler anywhere compiles this more slowly than just `T`, we can't have it.

Reply:
With just  `T` MSVC doesn't seem to propagate the __declspec(align()) attribute if I understand the issue correctly. Making it a const-ref bypasses this.